spring-cloud-gcp-starters/spring-cloud-gcp-starter-storage/src/main/java/org/springframework/cloud/gcp/storage/autoconfig/StorageAutoConfiguration.java

```diff
@@ -31,40 +30,42 @@
 import org.springframework.cloud.gcp.core.GcpScope;
 import org.springframework.cloud.gcp.storage.GcpStorageProperties;
 import org.springframework.cloud.gcp.storage.GoogleStorageProtocolResolver;
+import org.springframework.cloud.gcp.storage.GoogleStorageProtocolResolverContext;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
 
 /**
- * An auto-configuration for Google {@link Storage} bean definition. Also it
- * {@link Import} a {@link GoogleStorageProtocolResolver} to register it
- * with the {@code DefaultResourceLoader}.
+ * An auto-configuration for Google {@link GoogleStorageProtocolResolverContext} bean
+ * definition. Also it {@link Import} a {@link GoogleStorageProtocolResolver} to register
+ * it with the {@code DefaultResourceLoader}.
  *
  * @author Vinicius Carvalho
  * @author Artem Bilan
  *
  * @see GoogleStorageProtocolResolver
  */
 @Configuration
-@ConditionalOnClass(Storage.class)
+@ConditionalOnClass(GoogleStorageProtocolResolverContext.class)
 @EnableConfigurationProperties({GcpProperties.class, GcpStorageProperties.class})
 @Import(GoogleStorageProtocolResolver.class)
 public class StorageAutoConfiguration {
 
 	@Bean
 	@ConditionalOnMissingBean
-	public static Storage storage(
+	public static GoogleStorageProtocolResolverContext googleStorageProtocolResolverContext(
```

Comment:
It was nice being able to inject a `Storage` bean after adding the GCS starter to the app. Now, one will have to rely on the `GoogleStorageProtocolResolverContext` instead.

Reply:
I would just configure both beans.